src/specfun.jl

```diff
@@ -26,3 +26,12 @@ gamma(x::Float128) =
     Float128(@ccall(libquadmath.tgammaq(x::Cfloat128)::Cfloat128))
 lgamma(x::Float128) =
     Float128(@ccall(libquadmath.lgammaq(x::Cfloat128)::Cfloat128))
+
+function logabsgamma(x::Float128)
+    if isfinite(x)
+      sign = x >= 0 ? 1 : 2*mod(ceil(Int64,x),2)-1
+    else
+      sign = 1
+    end
```

Comment:
I'm not sure why I was requested as a reviewer here 😄 In any case, I wonder why not just
```suggestion
    sign = !isfinite(x) || x >= 0 || !iszero(mod(ceil(x), 2)) ? 1 : -1
```
I was a bit concerned whether `x::Float128` (or rather the next integer value) is always representable as a `Int64` - but maybe it is?

Note also that
- this implementation differs from SpecialFunctions since e.g. `logabsgamma(-2)[2] logabsgamma(-2.0)[2] = 1` (and not `-1`!). So at least an additional branch for negative integers seems missing here?
- `SpecialFunctions.lgamma` is deprecated and probably will be removed in a future release

Reply:
Well thanks for the review anyway.

I was just worried about the type of the sign being always `Int64`, as this is the case for `SpecialFunctions.logabsgamma`. In any case your method is more elegant and won't cause possible representability issues.

As for the sign on the negative integers, I thought it didn't matter, but now I've made it match the behaviour of `SpecialFunctions.logabsgamma`.

I'm not sure what we could do about the deprecation of `SpecialFunctions.lgamma`. We need to export the one from `libquadmath` anyway in order to define `logabsgamma` from it.

Reply:
> I'm not sure what we could do about the deprecation of SpecialFunctions.lgamma. We need to export the one from libquadmath anyway in order to define logabsgamma from it.

I think you could just use the default deprecated definition from SpecialFunctions (`lgamma(x::Real) = logabsgamma(x)[1]` https://github.com/JuliaMath/SpecialFunctions.jl/blob/c1ce5b1d91d0060b5f20fb653925453fe644e40e/src/deprecated.jl#L6) (but not define it for `Float128`!), and then use 
```julia
    result = Float128(@ccall(libquadmath.lgammaq(x::Cfloat128)::Cfloat128))
```
in `logabsgamma` in this PR.

Reply:
Oh that's a good point, we don't need to define it at all, since we have `logabsgamma` the default definition will work. Also when they do remove it there will be no need to change anything here. Done.